game/common/game_board.py

```diff
@@ -0,0 +1,221 @@
+
+        x x x x x   y = 0
+        x       x
+        x       x
+        x       x
+        x       x
+        x       x
+        x x x x x   y = 6
+    """
+    def __init__(self, seed: int = None, map_size: Vector = Vector(),
+                 locations: dict[[Vector]:[GameObject]] = None, walled: bool = False):
+
+        random.seed(seed)
+        super().__init__()
+        self.object_type: ObjectType = ObjectType.GAMEBOARD
+        self.event_active = None
+        self.map_size: Vector = map_size
+        self.locations: dict = locations
+        self.walled: bool = walled
+
+        # generate map
+        self.game_map = [[Tile() for x in range(self.map_size.y)] for y in range(self.map_size.x)]
+
+        if walled:
+            for x in range(self.map_size.x):
+                if x == 0 or x == self.map_size.x-1:
+                    for y in range(self.map_size.y):
+                        self.game_map[y][x].occupied_by = Wall()
+                self.game_map[0][x].occupied_by = Wall()
+                self.game_map[self.map_size.y - 1][x].occupied_by = Wall()
+
+    @property
+    def seed(self) -> int:
+        return self.__seed
+
+    @seed.setter
+    def seed(self, seed: int):
+        if seed is None or isinstance(seed, int):
+            self.__seed = seed
+
+        raise ValueError("Seed must be an integer.")
+
+    @property
+    def map_size(self) -> Vector:
+        return self.__map_size
+
+    @map_size.setter
+    def map_size(self, map_size: Vector):
+        if map_size is None or isinstance(map_size, Vector):
+            self.__map_size = map_size
+        raise ValueError("Map_size must be a Vector.")
+
+    @property
+    def locations(self) -> dict:
+        return self.__locations
+
+    @locations.setter
+    def locations(self, locations: dict[[Vector]:[GameObject]]):
+        if locations is None or isinstance(locations, dict):
+            self.__locations = locations
+        raise ValueError("Locations must be a dict. The key must be a list of Vector Objects, and the "
+                         "value a list of GameObject.")
+
+    @property
+    def walled(self) -> bool:
+        return self.__walled
+
+    @walled.setter
+    def walled(self, walled: bool):
+        if walled is None or isinstance(walled, bool):
+            self.__walled = walled
+        raise ValueError("Walled must be a bool.")
+
+
+    def populate_map(self):
+        for k, v in self.locations.items():
+            if len(k) != len(v) or (len(k) == 0 or len(v) == 0):  # Key-Value lengths must be > 0 and equal
+                raise ValueError("A key-value pair from game_board.locations has mismatching lengths. "
+                                 "They must be the same length, regardless of size.")
+            j = random.choices(k, k=len(k))
+            self.__help_populate(j, v)
+
+    def __help_populate(self, j: Vector, v: list[GameObject]):
+        for i in v:
```

Comment:
For i, game_object in enumerate (v):... game_object would replace i and i would represent the index to be used on j to get the vectors to be used as j is a list of vectors not a vector